app/Observers/CourseObserver.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Observers;
+
+use App\Models\Course;
+use Illuminate\Support\Str;
+
+class CourseObserver
+{
+    public function creating(Course $course): void
+    {
+        if ($course->slug) {
+            return;
+        }
+
+        $slug = Str::slug($course->abbreviation);
+
+        if (!$this->checkIfSlugExists($slug)) {
```

Comment:
Anyway, this is probably copypasted from news observer. Maybe it could be encapsulated into some separate class/trait?

Reply:
updated